java/src/main/java/api/example/java/controllers/HomeController.java

```diff
@@ -0,0 +1,21 @@
+package api.example.java.controllers;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@Controller
+public class HomeController extends BaseController {
+    private static final String LOGIN_URI = "loginUri";
+    @GetMapping("/")
+    public String home(Model model, HttpSession session, HttpServletRequest request) {
+        if (isUserLoggedIn(session)) {
+            return HOME_PAGE;
+        }
+        model.addAttribute(LOGIN_URI, config.buildOauthLink(request.getRequestURL().toString()));
```

Comment:
Rather than passing the current URI here as a string and hard-coding the extra "login-redirect" element in the Config class, it might be more clear to use another URI builder, like:
```java
ServletUriComponentsBuilder.fromRequestUri(request).pathSegment("login-redirect").build();
```

Reply:
And, optionally, including the "login-redirect" string in the BaseController class so it can be shared here and in the LoginController GetMapping annotation.

Reply:
done. But did not use the constant in the get mapping. Since getMapping needs '/' character in the front.